saleor/graphql/checkout/mutations/checkout_remove_promo_code.py

```diff
@@ -84,31 +84,28 @@ def perform_mutation(
             "promo_code", promo_code, "promo_code_id", promo_code_id
         )
 
-        object_type, promo_code_pk = cls.clean_promo_code_id(promo_code_id)
-
         checkout = get_checkout(cls, info, checkout_id=checkout_id, token=token, id=id)
 
         manager = get_plugin_manager_promise(info.context).get()
         checkout_info = fetch_checkout_info(checkout, [], manager)
 
-        removed = False
         if promo_code:
-            removed = remove_promo_code_from_checkout(checkout_info, promo_code)
+            remove_promo_code_from_checkout_or_error(checkout_info, promo_code)
         else:
-            removed = cls.remove_promo_code_by_id(
+            object_type, promo_code_pk = cls.clean_promo_code_id(promo_code_id)
+            cls.remove_promo_code_by_id_or_error(
                 info, checkout, object_type, promo_code_pk
             )
-
-        if removed:
-            lines, _ = fetch_checkout_lines(checkout)
-            invalidate_checkout(
-                checkout_info,
-                lines,
-                manager,
-                recalculate_discount=True,
-                save=True,
-            )
-            cls.call_event(manager.checkout_updated, checkout)
+        # if this step is reached, it means promo code was removed
```

Comment:
I'm not sure about that. There might be the case when the voucher code exists but it's not assigned to checkout, so we are not removing the promo code from checkout and not raising an error, right? In such case, we do not want to reach the below code.